components/provisioning/org.wso2.carbon.identity.provisioning/src/main/java/org/wso2/carbon/identity/provisioning/OutboundProvisioningManager.java

```diff
@@ -122,75 +123,87 @@ private static int getTenantIdOfDomain(String tenantDomain)
     private Map<String, RuntimeProvisioningConfig> getOutboundProvisioningConnectors(
             ServiceProvider serviceProvider, String tenantDomainName) throws IdentityProvisioningException {
 
+        if (serviceProvider == null) {
+            throw new IdentityProvisioningException("Given Service Provider is null hence Outbound Provisioning " +
+                    "Connectors cannot be retrieved for tenant:" + tenantDomainName);
+        }
+
+        if (CarbonContext.getThreadLocalCarbonContext() == null) {
+            throw new IdentityProvisioningException("ThreadLocalCarbonContext cannot be null");
+        }
+
         Map<String, RuntimeProvisioningConfig> connectors = new HashMap<>();
+        String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
 
-        // maintain the provisioning connector cache in the super tenant.
-        // at the time of provisioning there may not be an authenticated user in the system -
-        // specially in the case of in-bound provisioning.
 
-        String tenantDomain = null;
-        int tenantId = -1234;
-        ServiceProviderProvisioningConnectorCacheKey key = null;
-        ServiceProviderProvisioningConnectorCacheEntry entry = null;
+        connectors = getRuntimeProvisioningConfigFromCache(serviceProvider,
+                tenantDomainName, tenantDomain, tenantId);
+        if (!connectors.isEmpty()) {
+            return connectors;
```

Comment:
how about a debug log here saying we didn't find any connectors so we return?

Reply:
Thanks Fara. Idea of this pull request is to re-factor some methods without changing any logic. Shall we get this merge and address those concerns in different pull request ?